Update LUIS authoring SDK to reflect latest changes

AmirGeorge opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
<!--
Please add an informative description that covers that changes made by the pull request.

If you are regenerating your SDK based off of a new swagger spec, please add the link to the corresponding swagger spec pull request that has been merged in the azure-rest-api-specs repository
-->

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

AmirGeorge committed:
Update LUIS authoring SDK to reflect latest changes

yangyuan commented on src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Authoring/Generated/Models/ApplicationPublishObject.cs:
```diff
@@ -32,13 +32,10 @@ public ApplicationPublishObject()
         {
             VersionId = versionId;
             IsStaging = isStaging;
-            Region = region;
```
This (I guess) should be considered as a breaking change. Thus might require a major version bump.

AmirGeorge committed:
Major version upgrade to 3.0.0 due to possible breaking change in removing region parameter in publish

dsgouda commented:
@AmirGeorge please fix failing tests

AmirGeorge commented:
> @AmirGeorge please fix failing tests

@dsgouda any input on why the delete app API is now failing? https://azuresdkci.westus2.cloudapp.azure.com/job/CI-NetSDK_U/1065/console

The only change added to it was to add an optional boolean query parameter.

dsgouda commented:
@AmirGeorge A couple of things to check
- Please check if the delete operation URL was modified in the spec
- If not, you may have to re-record your tests

Also, do these tests work on your local machine?

AmirGeorge commented:
> @AmirGeorge A couple of things to check
> 
> * Please check if the delete operation URL was modified in the spec
> * If not, you may have to re-record your tests
> 
> Also, do these tests work on your local machine?

@dsgouda No they do not work on my local machine. The test project was even build failing locally and I had to add xunit reference in the project file (added in this PR) to get it to build. Now when I try to run the tests from VS test explorer, it instantly says 'Test run finished' without actually running anything.

dsgouda commented:
@AmirGeorge could you add the link to your REST spec PR and check the first bullet point

AmirGeorge commented:
> @AmirGeorge could you add the link to your REST spec PR and check the first bullet point

@dsgouda https://github.com/Azure/azure-rest-api-specs/commit/7d344e289c4befb3cf2d92baee7ea1af204faae8

The only change is the addition of the optional force flag (topmost change in the linked commit). 

Could the query param be somehow interpreted as part of the url path not as a query param? The error message suggests this:
Unable to find a matching HTTP request for URL 'DELETE /luis/api/v2.0/apps/0645d086-d6a3-4a4b-9a9d-5efaab7bbd71?force=false'.

dsgouda commented:
@AmirGeorge that's right,
The bool is a Path parameter as indicated [here](https://github.com/Azure/azure-rest-api-specs/commit/7d344e289c4befb3cf2d92baee7ea1af204faae8#diff-b30ca75196a51d0afe8d4bcf3d5f48d1R9368)
You may have to re-record the tests

AmirGeorge commented:
> @AmirGeorge that's right,
> The bool is a Path parameter as indicated [here](https://github.com/Azure/azure-rest-api-specs/commit/7d344e289c4befb3cf2d92baee7ea1af204faae8#diff-b30ca75196a51d0afe8d4bcf3d5f48d1R9368)
> You may have to re-record the tests

@dsgouda I am not sure how to do that. Any pointers?

dsgouda commented:
@AmirGeorge instructions to record tests can be found [here](https://github.com/Azure/azure-sdk-for-net/blob/master/Documentation/Using-Azure-TestFramework.md)

AmirGeorge committed:
manual fix for tests issue

dsgouda commented:
@AmirGeorge Please check if the tests work for you in PlayBack mode. Also, please ensure you have copied over the newly recorded test jsons to the SessionRecords directory

AmirGeorge committed:
add missing references in test project

AmirGeorge committed:
Revert "manual fix for tests issue"

This reverts commit 4ac8378546611613f0ffaad246f6e64f4df5b6dd.

AmirGeorge committed:
FIXING  AND RERECORDING TESTS!

AmirGeorge commented:
@dsgouda finally done.

FYI the link to record tests was a bit confusing. It was not clear whether I should start from creating a service principal or what. I ended up just changing the `Playback` mode in `BaseTest` to `Record` and assign a valid subscription key.

It took a lot of time because at first the tests project was failing to build locally, so I added the xunit nuget package:
`<PackageReference Include="xunit" Version="2.4.1" />`

After this it could build, but on trying to run the tests from tests explorer, the test run exists instantly without actually running anything. Doing some research, I arrived that I need to add the following 2 references:
`<PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />`
`<PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">`

After that the tests worked in playback mode. I changed to record and re-recorded all the tests with many fixes to get them working.

dsgouda commented:
> 
> 
> @dsgouda finally done.
> 
> FYI the link to record tests was a bit confusing. It was not clear whether I should start from creating a service principal or what. I ended up just changing the `Playback` mode in `BaseTest` to `Record` and assign a valid subscription key.
> 
> It took a lot of time because at first the tests project was failing to build locally, so I added the xunit nuget package:
> `<PackageReference Include="xunit" Version="2.4.1" />`
> 
> After this it could build, but on trying to run the tests from tests explorer, the test run exists instantly without actually running anything. Doing some research, I arrived that I need to add the following 2 references:
> `<PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />`
> `<PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">`
> 
> After that the tests worked in playback mode. I changed to record and re-recorded all the tests with many fixes to get them working.

That's interesting, @shahabhijeet has a better handle on test dependencies and can investigate the issue you faced. For now, I can review rest of the PR since the checks are passing.
Thanks for the updates

dsgouda commented on src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Authoring.Tests/Microsoft.Azure.CognitiveServices.LUIS.Authoring.Tests.csproj:
```diff
@@ -9,6 +9,12 @@
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
+    <PackageReference Include="xunit" Version="2.4.1" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">
+      <PrivateAssets>all</PrivateAssets>
```
Do you need to add the `PrivateAssets` and `IncludeAssets` tags?

AmirGeorge replied:
@dsgouda done.

dsgouda reviewed (changes requested):
Looks good for the most part

AmirGeorge committed:
removing private assets and include assets

dsgouda reviewed (approved):
LGTM
